<!doctype html>
<!--[if lt IE 7]><html class="no-js ie6 oldie" lang="ja"><![endif]-->
<!--[if IE 7]>   <html class="no-js ie7 oldie" lang="ja"><![endif]-->
<!--[if IE 8]>   <html class="no-js ie8 oldie" lang="ja"><![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="ja"><!--<![endif]-->
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<title>vcfExtract</title>
	<meta name="description" content="Bioinformatics tutorials, Linux, HPC, NGS, NGS workflow manager, TOGGLe, galaxy">
	<meta name="description" content="Commands to manipulate VCF files">
	
	<meta name="author" content=" Sébastien RAVEL -- sebastien.ravel@cirad.fr">
	
	<meta name="author" content=" Christine Tranchant-Dubreuil -- christine.tranchant@ird.fr">
	

	<meta name="viewport" content="width=device-width,initial-scale=1">

	<link rel="stylesheet" type="text/css" media="screen" href="http://toggle.ird.fr:4000/css/bootstrap.min.css" >
	<link rel="stylesheet" type="text/css" media="screen" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
	<link rel="stylesheet" type="text/css" media="screen" href="http://toggle.ird.fr:4000/css/monokai.css">
	<link rel="stylesheet" type="text/css" media="screen" href="http://toggle.ird.fr:4000/css/stylesheet.css">
	<link rel="icon" href="http://toggle.ird.fr:4000/images/southgreenlong.png">
</head>
 <body>

	<!-- Static navbar -->
	<nav class="navbar navbar-default navbar-inverse navbar-static-top">
		<div class="container">
		<div class="navbar-header">
			<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
			<span class="sr-only"></span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
			</button>
			<a class="navbar-brand visible-xs" href="#"></a>
		</div>
		<div id="navbar" class="navbar-collapse collapse">
			<ul id="menu">
				<li><a href="http://toggle.ird.fr:4000/">Home</a></li>
				<li><a href="#">Bioanalysis</a><span class="caret" ></span>
					<ul id="child">
						<li><a href="http://toggle.ird.fr:4000/bioanalysis/polymorphism/">Polymorphism detection</a></li>
						<li><a href="http://toggle.ird.fr:4000/bioanalysis/vcfExtract/">VCF manipulate</a></li>
						<li><a href="http://toggle.ird.fr:4000/bioanalysis/rnaSeq/">RNA-Seq</a></li>
					</ul>
				</li>
				<li><a href="#">Cluster</a><span class="caret" ></span>
					<ul id="child">
						<li><a href="http://toggle.ird.fr:4000/cluster/HPCConnection/">HPC connection</a></li>
						<li><a href="http://toggle.ird.fr:4000/cluster/SGE/">SGE</a></li>
						<li><a href="http://toggle.ird.fr:4000/cluster/moduleLoad/">Use module system</a></li>
					</ul>
				</li>
				<li><a href="#">Toolbox&Trick</a><span class="caret" ></span>
					<ul id="child">
						<li><a href="http://toggle.ird.fr:4000/toolbox/linux/">Linux</a></li>
						<li><a href="http://toggle.ird.fr:4000/toolbox/git/">Git</a></li>
						<li><a href="http://toggle.ird.fr:4000/toolbox/docker/">Docker</a></li>
					</ul>
				</li>
				<li><a href="http://toggle.ird.fr:4000/contact/">About Us</a></li>
				<li><a id="forkme_banner" href="https://github.com/SouthGreenPlatform/trainings/">GitHub</a></li>
			</ul>
		</div>
		</div>
	</nav>
	<!--/.nav-collapse -->
	<!-- Header -->
	<div id="header_wrap" class="outer">
		<header class="inner">
		<div class="header"><img class="img-responsive" width="25%" src="http://toggle.ird.fr:4000/images/southgreenlong.png" alt="South Green Logo" /></div>
		<h2 id="project_tagline">South Green tutorials Web Site</h2>
		</header>
	</div>

<body>
<div id="main_content_wrap" class="outer">
	 <section id="main_content" class="inner">
<table class="table-contact">
<tr>
<td><img width="100%" src="http://toggle.ird.fr:4000/images/trainings-linux.png" alt="" />
</td>
<td>
<h1> Commands to manipulate VCF files</h1><br />
This page describes a serie of tools and linux commands used to manipulate VCF files.
</td>
</tr>
</table>

<p>We need, in this tutorial:</p>
<ul>
  <li>1 vcf file</li>
  <li>GATK tools</li>
  <li>bcftools</li>
</ul>

<h3 id="authors">Author(s)</h3>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Authors</th>
      <th style="text-align: left">Christine Tranchant-Dubreuil</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Research Unit</td>
      <td style="text-align: left">UMR DIADE</td>
    </tr>
    <tr>
      <td style="text-align: left">Institut</td>
      <td style="text-align: left"><img src="https://www.ird.fr/extension/ird/design/ird/images/picto/logo_ird.png" width="20%" /></td>
    </tr>
  </tbody>
</table>

<h4 id="keywords--gatkbcftools"><em>Keywords</em> : <code class="highlighter-rouge">gatk</code>,<code class="highlighter-rouge">bcftools</code></h4>

<h4 id="files-format--vcf"><em>Files format</em> : vcf</h4>

<h4 id="date--10032017"><em>Date</em> : 10/03/2017</h4>

<hr />

<h2 id="summary">Summary</h2>

<!-- TOC depthFrom:2 depthTo:2 withLinks:1 updateOnSave:1 orderedList:0 -->

<ul>
  <li><a href="#sample-list">Extracting list of samples from a vcf file with <code class="highlighter-rouge">grep</code>and <code class="highlighter-rouge">cut</code>commands</a></li>
  <li><a href="#sample-from-vcf-gatk">Extracting a subset of samples from a multigenome vcf file with <code class="highlighter-rouge">GATK selectVariants</code></a></li>
  <li><a href="#sample-from-vcf-bcftools">Extracting a subset of samples from a multigenome vcf file with <code class="highlighter-rouge">bcftools</code></a></li>
  <li><a href="#calculating-pi">Calculating the nucleotide diversity from a vcf file with <code class="highlighter-rouge">vcftools</code></a></li>
</ul>

<!-- /TOC -->

<hr />

<p><a name="sample-list"></a></p>
<h2 id="extracting-list-of-samples-from-a-vcf-file-with-grepand-cutcommands">Extracting list of samples from a vcf file with <code class="highlighter-rouge">grep</code>and <code class="highlighter-rouge">cut</code>commands</h2>

<p>one line with all samples</p>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">grep</span> <span class="s2">"#CHROM"</span> <span class="n">output</span> <span class="o">|</span> <span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">10</span><span class="o">-</span></code></pre></figure>

<p>one line by sample</p>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">grep</span> <span class="s2">"#CHROM"</span> <span class="n">output</span> <span class="o">|</span> <span class="n">cut</span> <span class="o">-</span><span class="n">f</span> <span class="mi">10</span><span class="o">-</span> <span class="o">|</span> <span class="n">xargs</span> <span class="o">-</span><span class="n">n</span> <span class="mi">1</span></code></pre></figure>

<p><a name="sample-from-vcf-gatk"></a></p>
<h2 id="extracting-a-subset-of-samples-from-a-multigenome-vcf-file-with-gatk-selectvariants">Extracting a subset of samples from a multigenome vcf file with <code class="highlighter-rouge">GATK selectVariants</code></h2>

<h4 id="select-two-samples-out-of-a-vcf-with-many-samples">Select two samples out of a vcf with many samples</h4>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">java</span> <span class="o">-</span><span class="no">Xmx12g</span> <span class="o">-</span><span class="n">jar</span> <span class="sr">/usr/</span><span class="n">local</span><span class="o">/</span><span class="n">gatk</span><span class="o">-</span><span class="mf">3.6</span><span class="o">/</span><span class="no">GenomeAnalysisTK</span><span class="p">.</span><span class="nf">jar</span> <span class="o">-</span><span class="no">T</span> <span class="no">SelectVariants</span> <span class="o">-</span><span class="no">R</span> <span class="n">reference</span><span class="p">.</span><span class="nf">fa</span> <span class="o">-</span><span class="no">V</span> <span class="n">inputFileName</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">-</span><span class="n">o</span> <span class="n">outputFilename</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">-</span><span class="n">sn</span> <span class="n">sample1</span> <span class="o">-</span><span class="n">sn</span> <span class="n">sample2</span></code></pre></figure>

<h6 id="rk--if-you-get-the-following-error-message-fasta-dict-file--for-reference--does-not-exist-please-see-httpswwwbroadinstituteorggatkguidearticleid1601-for-help-creating-in">Rk : if you get the following error message “<em>Fasta dict file … for reference … does not exist</em>”, please see https://www.broadinstitute.org/gatk/guide/article?id=1601 for help creating in.</h6>

<h4 id="select-genotypes-from-a-file-containing-a-list-of-samples-to-include">Select genotypes from a file containing a list of samples to include</h4>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">java</span> <span class="o">-</span><span class="no">Xmx12g</span> <span class="o">-</span><span class="n">jar</span> <span class="sr">/usr/</span><span class="n">local</span><span class="o">/</span><span class="n">gatk</span><span class="o">-</span><span class="mf">3.6</span><span class="o">/</span><span class="no">GenomeAnalysisTK</span><span class="p">.</span><span class="nf">jar</span> <span class="o">-</span><span class="no">T</span> <span class="no">SelectVariants</span> <span class="o">-</span><span class="no">R</span> <span class="n">reference</span><span class="p">.</span><span class="nf">fa</span> <span class="o">-</span><span class="no">V</span> <span class="n">inputFileName</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">-</span><span class="n">o</span> <span class="n">outputFileName</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">--</span><span class="n">sample_file</span> <span class="n">barthii</span><span class="p">.</span><span class="nf">only</span><span class="o">.</span><span class="no">RG</span><span class="p">.</span><span class="nf">list</span>  <span class="o">--</span><span class="no">ALLOW_NONOVERLAPPING_COMMAND_LINE_SAMPLES</span></code></pre></figure>

<h4 id="select-genotypes-from-a-file-containing-a-list-of-samples-to-exclude">Select genotypes from a file containing a list of samples to exclude</h4>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">java</span> <span class="o">-</span><span class="no">Xmx12g</span> <span class="o">-</span><span class="n">jar</span> <span class="sr">/usr/</span><span class="n">local</span><span class="o">/</span><span class="n">gatk</span><span class="o">-</span><span class="mf">3.6</span><span class="o">/</span><span class="no">GenomeAnalysisTK</span><span class="p">.</span><span class="nf">jar</span> <span class="o">-</span><span class="no">T</span> <span class="no">SelectVariants</span> <span class="o">-</span><span class="no">R</span> <span class="n">reference</span><span class="p">.</span><span class="nf">fa</span> <span class="o">-</span><span class="no">V</span> <span class="n">inputFileName</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">-</span><span class="n">o</span> <span class="n">outputFileName</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">--</span><span class="n">exclude_sample_file</span> <span class="n">barthii</span><span class="p">.</span><span class="nf">only</span><span class="o">.</span><span class="no">RG</span><span class="p">.</span><span class="nf">list</span>  <span class="o">--</span><span class="no">ALLOW_NONOVERLAPPING_COMMAND_LINE_SAMPLES</span></code></pre></figure>

<h6 id="rk--if-you-get-the-following-error-message--bad-input-samples-entered-on-command-line-through--sf-or--sn-that-are-not-present-in-the-vcf-run-with-allow_nonoverlapping_command_line_samples">Rk : if you get the following error message : “<em>Bad input: Samples entered on command line (through -sf or -sn)) that are not present in the VCF</em>”, run with –ALLOW_NONOVERLAPPING_COMMAND_LINE_SAMPLES</h6>

<p><a name="sample-from-vcf-bcftools"></a></p>
<h2 id="extracting-a-subset-of-samples-from-a-multigenome-vcf-file-with-bcftools">Extracting a subset of samples from a multigenome vcf file with <code class="highlighter-rouge">bcftools</code></h2>
<h4 id="select-genotypes-from-a-file-containing-a-list-of-samples-to-include-with-bcftools">Select genotypes from a file containing a list of samples to include with <code class="highlighter-rouge">bcftools</code></h4>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">bcftools</span> <span class="n">view</span> <span class="o">-</span><span class="no">S</span> <span class="n">barthii</span><span class="p">.</span><span class="nf">only</span><span class="o">.</span><span class="no">RG</span><span class="p">.</span><span class="nf">list</span> <span class="n">inputFileName</span><span class="p">.</span><span class="nf">vcf</span> <span class="o">--</span><span class="n">force</span><span class="o">-</span><span class="n">samples</span> <span class="o">-</span><span class="n">o</span> <span class="n">outputFilename</span><span class="p">.</span><span class="nf">vcf</span><span class="sb">`</span></code></pre></figure>

<p><a name="calculating-pi"></a></p>
<h2 id="calculating-the-nucleotide-diversity-from-a-vcf-file-with-vcftools">Calculating the nucleotide diversity from a vcf file with <code class="highlighter-rouge">vcftools</code></h2>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">vcftools</span> <span class="o">--</span><span class="n">vcf</span> <span class="n">inputFilename</span><span class="p">.</span><span class="nf">vcf</span>  <span class="o">--</span><span class="n">out</span> <span class="n">outputFilename</span><span class="o">.</span><span class="no">PI</span>  <span class="o">--</span><span class="n">window</span><span class="o">-</span><span class="n">pi</span> <span class="mi">100000</span> <span class="o">--</span><span class="n">remove</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">all</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-ruby" data-lang="ruby"><span class="n">grep</span> <span class="s2">"PI"</span> <span class="no">OgOb</span><span class="o">-</span><span class="n">all</span><span class="o">-</span><span class="no">MSU7</span><span class="o">-</span><span class="no">CHR2</span><span class="o">.</span><span class="no">GATKSV</span><span class="o">.</span><span class="no">VCFTOOLS</span><span class="p">.</span><span class="nf">stats</span><span class="o">-</span><span class="mi">100000</span><span class="p">.</span><span class="nf">windowed</span><span class="p">.</span><span class="nf">pi</span> <span class="o">-</span><span class="n">v</span> <span class="o">|</span> <span class="n">awk</span> <span class="s1">'{ sum+=$5; print $5,"; ",sum , "* ", NR ; } END { print "PI average :", sum / NR; }'</span></code></pre></figure>


<!--
<aside>
	<h3>aside</h3>
	<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam sodales urna non odio egestas tempor. Nunc vel vehicula ante. Etiam bibendum iaculis libero, eget molestie nisl pharetra in. In semper consequat est, eu porta velit mollis nec. Curabitur posuere enim eget turpis feugiat tempor. Etiam ullamcorper lorem dapibus velit suscipit ultrices.</p>
</aside>-->

	</div> <!-- #main -->
</div> <!-- #main-container -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="http://toggle.ird.fr:4000/js/jquery-1.11.3.min.js"></script>
    <script src="http://toggle.ird.fr:4000/js/bootstrap.min.js"></script>
    <script src="http://toggle.ird.fr:4000/js/homepage.js"></script>
    <script src="http://toggle.ird.fr:4000/js/libs/modernizr-2.0.6.min.js"></script>

</script>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <div id="footer">
			<div class="logo-div" class="clearfix">
				<a class="logo" href="http://www.ird.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_ird.png" alt="IRD Logo"  width="" />
				</a>

				<a class="logo" href="http://www.cirad.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo-cirad.png" alt="CIRAD Logo"   width="" />
				</a>

				<a class="logo" href="http://www.inra.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_inra.png" alt="INRA Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.bioversityinternational.org/">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_bioversity.png" alt="Bioversity Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.supagro.fr/">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_supagro.png" alt="SupAgro Logo"   width=""/>
				</a>
				
				<a class="logo" href="http://www.southgreen.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/southgreenlong.png" alt="South Green Logo"   width=""/>
				</a>

			</div>
		</div>

		<div class="footer_txt">
			<p>The contents of this website are &copy;&nbsp;2018 under the terms of the Licencied under
				<a href="http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html">CeCill-C</a>
				 and <a href="/blob/master/LICENSE">GPLv3</a>.
			</p>
		</div>
      </footer>
    </div>

<script src="//ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
<script>window.jQuery || document.write('<script src="js/libs/jquery-1.7.1.min.js"><\/script>')</script>

<!--<script src="http://toggle.ird.fr:4000/js/script.js"></script>
<script>
	var _gaq=[['_setAccount',''],['_trackPageview']];
	(function(d,t){var g=d.createElement(t),s=d.getElementsByTagName(t)[0];g.async=1;
	g.src=('https:'==location.protocol?'//ssl':'//www')+'.google-analytics.com/ga.js';
	s.parentNode.insertBefore(g,s)}(document,'script'));
</script>

<!--[if lt IE 7 ]>
	<script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.2/CFInstall.min.js"></script>
	<script>window.attachEvent("onload",function(){CFInstall.check({mode:"overlay"})})</script>
<![endif]-->

</body>
</html>
